keras/layers/wrappers.py

```diff
@@ -163,30 +163,47 @@ def compute_output_shape(self, input_shape):
         child_input_shape = (input_shape[0],) + input_shape[2:]
         child_output_shape = self.layer.compute_output_shape(child_input_shape)
         timesteps = input_shape[1]
-        return (child_output_shape[0], timesteps) + child_output_shape[1:]
+        if isinstance(child_output_shape, tuple):
+            return (child_output_shape[0], timesteps) + child_output_shape[1:]
+        elif isinstance(child_output_shape, list):
+            return [(shape[0], timesteps) + shape[1:] for shape in child_output_shape]
+        msg = 'output shape should be a tuple (single output) or a list (multiple outputs)'
+        raise ValueError(msg)
 
     def call(self, inputs, training=None, mask=None):
+        class LPWrapper:
+            # hack equivalent to nonlocal keyword
+            # This design was chosen for Python 2.7 compatibility.
+            uses_learning_phase = False
+
         kwargs = {}
         if has_arg(self.layer.call, 'training'):
             kwargs['training'] = training
-        uses_learning_phase = False
 
         input_shape = K.int_shape(inputs)
         if input_shape[0]:
             # batch size matters, use rnn-based implementation
             def step(x, _):
-                global uses_learning_phase
                 output = self.layer.call(x, **kwargs)
-                if hasattr(output, '_uses_learning_phase'):
-                    uses_learning_phase = (output._uses_learning_phase or
-                                           uses_learning_phase)
+                if not isinstance(output, list):
+                    output = [output]
+                uses_learning_phase = []
```

Comment:
You can simplify your code a lot by simply updating your boolean, instead of using a list of booleans.